helm/charts/hydra/templates/job-mirgation.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "hydra.fullname" . }}-automigrate
+  {{- if .Release.Namespace }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
+  labels:
+{{ include "hydra.labels" . | indent 4 }}
+    {{- with .Values.deployment.labels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- with .Values.deployment.annotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  template:
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}-automigrate
+        image: {{ .Values.image.repository }}:{{ include "hydra.getImageAlpine" . }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -ex
+          apk add curl 
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+          chmod +x kubectl
+          mv kubectl /usr/local/bin/kubectl
+          [[ -f /etc/migration/migration.json ]] && echo "Migration was done, nothing to do here" && exit 0
+          hydra migrate sql -e --yes --config /etc/config/config.yaml
+          echo {\"timestamp\": \"$(date "+%Y%m%d-%H%M%S")\",\"completed\": true} > /tmp/migration.json
+          kubectl create configmap {{ include "hydra.fullname" . }}-migration --from-file=/tmp/migration.json
```

Comment:
This `kubectl create` should be replaced with ``--dry-run | kubectl apply``` IMO as job will be executed per install/upgrade and will fail with with already exists error from kubectl.

And since this will be part of helm release (Job) won't it cause troubles as images in jobs are immutable? Ideally it should be a hook but probably the config volume CM is not on a hook as well?

Reply:
My idea was to delete the cm if it already exists (so, upgrade scenario), which would block new instances from running too early. 
The job will be on pre-install,pre-uprade hooks, just didn't add them yet to better debug locally ;)

Reply:
How this will block new instances? I see you do the `test -f && k delete` but this will always delete the cm?